src/main/java/ru/mercuriev/game/of/life/graph/cell/Cell.java

```diff
@@ -0,0 +1,201 @@
+package ru.mercuriev.game.of.life.graph.cell;
+
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * This class represents both Graph and its element - Cell
```

Comment:
Yes, this may seem not clear. But since Cell is recursive structure it does not need special class that holds the pointer to the *main* cell. May be you wil agree for comments to be changed?